x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/RemoteClusterSecurityCrossClusterApiKeySigningIT.java

```diff
@@ -0,0 +1,230 @@
+            .setting("xpack.security.remote_cluster_server.ssl.certificate", "remote-cluster.crt")
+            .configFile("signing_ca.crt", Resource.fromClasspath("signing/root.crt"))
+            .setting("cluster.remote.signing.certificate_authorities", "signing_ca.crt")
+            .keystore("xpack.security.remote_cluster_server.ssl.secure_key_passphrase", "remote-cluster-password")
+            .build();
+
+        queryCluster = ElasticsearchCluster.local()
+            .name("query-cluster")
+            .apply(commonClusterConfig)
+            .setting("xpack.security.remote_cluster_client.ssl.enabled", "true")
+            .setting("xpack.security.remote_cluster_client.ssl.certificate_authorities", "remote-cluster-ca.crt")
+            .configFile("signing.crt", Resource.fromClasspath("signing/signing.crt"))
+            .setting("cluster.remote.my_remote_cluster.signing.certificate", "signing.crt")
+            .configFile("signing.key", Resource.fromClasspath("signing/signing.key"))
+            .setting("cluster.remote.my_remote_cluster.signing.key", "signing.key")
+            .keystore("cluster.remote.my_remote_cluster.credentials", () -> {
+                if (API_KEY_MAP_REF.get() == null) {
+                    final Map<String, Object> apiKeyMap = createCrossClusterAccessApiKey("""
+                        {
+                            "search": [
+                              {
+                                "names": ["index*", "not_found_index"]
+                              }
+                            ]
+                        }""");
+                    API_KEY_MAP_REF.set(apiKeyMap);
+                }
+                return (String) API_KEY_MAP_REF.get().get("encoded");
+            })
+            .keystore("cluster.remote.invalid_remote.credentials", randomEncodedApiKey())
+            .build();
+    }
+
+    @ClassRule
+    // Use a RuleChain to ensure that fulfilling cluster is started before query cluster
+    public static TestRule clusterRule = RuleChain.outerRule(fulfillingCluster).around(queryCluster);
+
+    public void testCrossClusterSearchWithCrossClusterApiKeySigning() throws Exception {
+        indexTestData();
+        assertCrossClusterSearchSuccessfulWithResult();
+
+        // Change the CA to something that doesn't trust the signing cert
+        updateClusterSettingsFulfillingCluster(
+            Settings.builder().put("cluster.remote.signing.certificate_authorities", "transport-ca.crt").build()
+        );
+        assertCrossClusterAuthFail();
+
+        // Update settings on query cluster to ignore unavailable remotes
+        updateClusterSettings(Settings.builder().put("cluster.remote.my_remote_cluster.skip_unavailable", Boolean.toString(true)).build());
+
+        assertCrossClusterSearchSuccessfulWithoutResult();
+
+        // TODO add test for certificate identity configured for API key but no signature provided (should 401)
```

Comment:
When the cross cluster api key with a cert identity is available these test will be implemented here.

Reply:
I'll do this in a separate PR to keep scope manageable.